### ISSUE TITLE
Remove detailed `SyntaxError` messages from `JSON.parse`

### ISSUE THREAD

**Issue by ghost** [bestiejs/json3#5]
**TL;DR: `JSON.parse` and `JSON.stringify` currently provide detailed error messages. This behavior is implementation-dependent, and increases the file size. Shall I remove these messages?**

Currently, the `JSON.parse` implementation provides very specific syntax error messages—"Illegal octal literal," "Invalid {Unicode} escape sequence in string," "Expected end-of-file," ""A single colon (`:`) must separate each object property name from the value," and so on (`JSON.stringify` also emits the message, "Cyclic structures cannot be serialized"). Unfortunately, this increases the file size, for a potentially inconsequential gain.

I can't imagine that this feature is well-known, and native `parse` implementations are typically much less specific: Chrome uses the catch-all "Unexpected end of input" and "Unexpected token: X" messages for many errors, Safari attempts to provide some additional details, and Firefox falls back to "Unexpected character." Likewise, JSON 2 only emits a generic syntax error if the JSON source string cannot be parsed. It's clear that relying on implementation-dependent error messages is unwise.

The only use case that I can contrive for the current behavior is a primitive syntax validator, though, because JSON 3 throws the first syntax error that it encounters, it's not particularly useful.

Removing the messages will reduce the code size to 3 KB (minified and `gzip`-ed), which was the size of v3.0. Without `gzip`, the size of the minified version will be reduced from 8.8 KB to 7.9 KB. The difference isn't drastic, but certainly noticeable, especially given the fact that JSON 2 is 1.3 KB.

Would love to obtain some feedback on this. Should the detailed error messages be retained, or can they be removed in the next point release (v3.2.1)?
